cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -672,13 +672,14 @@ private boolean deployAppAllowed(@Nullable String parentVersion,
 
   public void deleteApplication(String namespaceId, String appId, String versionId)
       throws IOException {
+    getApplicationEditTable().delete(getNamespaceApplicationKeys(new ApplicationReference(namespaceId, appId)));
```

Comment:
this shouldn't be in the version specific delete.

Reply:
created CDAP-20425